src/traces/funnel/attributes.js

```diff
@@ -0,0 +1,85 @@
+/**
+* Copyright 2012-2019, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var pieAtts = require('../pie/attributes');
+var barAttrs = require('../bar/attributes');
+var lineAttrs = require('../scatter/attributes').line;
+var extendFlat = require('../../lib/extend').extendFlat;
+var Color = require('../../components/color');
+
+module.exports = {
+    x: barAttrs.x,
+    x0: barAttrs.x0,
+    dx: barAttrs.dx,
+    y: barAttrs.y,
+    y0: barAttrs.y0,
+    dy: barAttrs.dy,
+
+    hovertext: barAttrs.hovertext,
+    hovertemplate: barAttrs.hovertemplate,
+
+    textinfo: extendFlat({}, pieAtts.textinfo, {
+        editType: 'plot',
+        flags: ['label', 'text', 'percent initial', 'percent previous', 'percent total', 'value'],
+        description: [
+            'Determines which trace information appear on the graph.'
+        ].join(' ')
+    }),
+
+    text: barAttrs.text,
+    textposition: extendFlat({}, barAttrs.textposition, {dflt: 'auto'}),
+    insidetextanchor: extendFlat({}, barAttrs.insidetextanchor, {dflt: 'middle'}),
+    insidetextrotate: extendFlat({}, barAttrs.insidetextrotate, {dflt: 'none'}),
+    textfont: barAttrs.textfont,
+    insidetextfont: barAttrs.insidetextfont,
+    outsidetextfont: barAttrs.outsidetextfont,
+    constraintext: barAttrs.constraintext,
+
+    cliponaxis: barAttrs.cliponaxis,
+    orientation: barAttrs.orientation,
+
+    offset: barAttrs.offset,
+    width: barAttrs.width,
+
+    marker: barAttrs.marker,
+
+    connector: {
+        fillcolor: {
+            valType: 'color',
+            role: 'style',
+            editType: 'plot', // TODO: could we set this to 'style' ?
```

Comment:
Yes, but make sure to add a jasmine test to :lock: the behaviour down.

Reply:
Revised in https://github.com/plotly/plotly.js/pull/3817/commits/4c741c7f477d53033ec320cbbb52c9c8c1d33f1d. Also regarding the test coverage we have these two blocks: https://github.com/plotly/plotly.js/blob/4c741c7f477d53033ec320cbbb52c9c8c1d33f1d/test/jasmine/tests/funnel_test.js#L1132-L1204.

Reply:
Thanks. Resolved.